client/network/src/protocol.rs

```diff
@@ -1202,21 +1215,45 @@ impl<B: BlockT, H: ExHashT> Protocol<B, H> {
 			return;
 		}
 		if let Some(transaction) = self.transaction_pool.transaction(hash) {
-			let propagated_to = self.do_propagate_transactions(&[(hash.clone(), transaction)]);
+			let propagated_to = self.do_propagate_transactions(
+				&[(hash.clone(), transaction)],
+				FloodSettings::Sqrt { minimum: 5 },
+			);
 			self.transaction_pool.on_broadcasted(propagated_to);
 		}
 	}
 
 	fn do_propagate_transactions(
 		&mut self,
 		transactions: &[(H, B::Extrinsic)],
+		flood_settings: FloodSettings,
 	) -> HashMap<H, Vec<String>> {
+		use rand::seq::IteratorRandom;
+
 		let mut propagated_to = HashMap::new();
-		for (who, peer) in self.context_data.peers.iter_mut() {
-			// never send transactions to the light node
-			if !peer.info.roles.is_full() {
-				continue;
-			}
+
+		let (selected_peers, limit) = match flood_settings {
+			FloodSettings::Everyone => (
+				self.context_data.peers
+					.iter()
+					.filter(|(_, peer)| peer.info.roles.is_full())
+					.map(|(id, _)| id.clone())
+					.collect::<Vec<_>>(),
+				None,
+			),
+			FloodSettings::Sqrt { minimum } => (
+				self.context_data.peers
+					.iter()
+					.filter(|(_, peer)| peer.info.roles.is_full())
```

Comment:
@arkpar was saying, that we have to include peers with authority role into the set for propagation anyway